scipy/stats/tests/test_odds_ratio.py

```diff
@@ -0,0 +1,61 @@
+import pytest
+import numpy as np
+from numpy.testing import assert_equal, assert_allclose
+from scipy.stats import nchypergeom_fisher, hypergeom
+from scipy.stats._odds_ratio import odds_ratio
+from .fisher_exact_results_from_r import data
+
+
+class TestOddsRatio:
+
+    @pytest.mark.parametrize('parameters, rresult', data)
+    def test_results_from_r(self, parameters, rresult):
+        alternative = parameters.alternative.replace('.', '-')
+        result = odds_ratio(parameters.table, alternative=alternative)
+        # The results computed by R are not very precise.
```

Comment:
They are very precise - 15 digits are included. Do you mean they are not very accurate? How do we know that the R results are inaccurate and `odds_ratio` is?

Reply:
Yes, that should be "... not very accurate."  The comment is based on lots of scratch work with mpmath, and on verifying the odds ratio with tests such as the self-check a few lines down from here.

I'm adding some additional consistency checks for the odds ratio confidence interval, and I'll add the odds ratio calculations to [`mpsci`](https://github.com/WarrenWeckesser/mpsci) so an mpmath-based version of the calculation is available.